gallifreyan.js

```diff
@@ -88,6 +89,24 @@ function updateText() {
     generateWords(words);
 }
 
+//discard any unfinished manual actions (line addition/deletion)
+function resetModes() {
+    deleteLineMode  = false;
+    if (addLineMode) {
+        addLineMode = false;
+        deleteLine(selectedLine);
+        selectedLine = null;
+    }
+}
+
+//create a new line and let user to position it
+function addNewLine() {
+    addLineMode = true;
+    selectedLine = new Line(allCircles[0], -PI/2, allCircles[0], -PI/2);
+    allCircles[0].lines.removeItem(selectedLine);
```

Comment:
It's gone. That was just a mistake. Yes, this commit is more hack-ish, I hope it won't cause any harm, but I can't come up with any better/cleaner solution.